x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeField.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import org.elasticsearch.common.time.IsoLocale;
+import org.elasticsearch.xpack.sql.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public interface DateTimeField {
+
+    static <D extends DateTimeField> Map<String, D> initializeResolutionMap(D[] values) {
+        Map<String, D> nameToPart;
```

Comment:
Why not a single line `Map<String, D> nameToPart = new HashMap<>();` instead of two lines?